index.js

```diff
@@ -130,10 +131,13 @@ class Schematic {
 
   static async read (buffer, version = null) {
     const schem = nbt.simplify(await parseNbt(buffer))
-    try {
-      return sponge.read(schem, version)
-    } catch {
-      return mcedit.read(schem, version)
+    const formats = [sponge, spongeV3, mcedit]
+    for (const format of formats) {
+      try {
+        return format.read(schem, version)
+      } catch (e) {
+
+      }
     }
```

Comment:
It would be good if it throws an error if it can't read the format. Right now, it could return undefined if the format is unrecognized. This is wrong for the type definition and also bad if people are wondering why they get undefined instead of a schematic or error.

Reply:
@DanielSchmerber

Reply:
What I don't like is that all errors, even those unrelated to the format, are discarded here in favor of a generic error message that doesn't say anything about what triggered the error. There should also really be a parameter passed here to specify the schematic type, that can fallback to trying different types.